libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/LargeCausalIndividualView/LargeCausalIndividualChart.tsx

```diff
@@ -0,0 +1,326 @@
+      [],
+      this.state.isLocalCausalDataLoading,
+      true,
+      false,
+      this.selectPointFromChartLargeData
+    );
+    this.setState({
+      chartProps,
+      isBubbleChartRendered: false,
+      isRevertButtonClicked: false,
+      plotData: datasetBarConfigOverride
+    });
+  };
+
+  private onBubbleClick = (
+    scatterPlotData: IHighchartsConfig,
+    xSeries: number[],
+    ySeries: number[],
+    indexSeries: number[]
+  ): void => {
+    this.setState({
+      indexSeries,
+      isBubbleChartRendered: false,
+      plotData: scatterPlotData,
+      xSeries,
+      ySeries
+    });
+  };
+
+  private onXSet = (value: ISelectorConfig): void => {
+    if (this.state.chartProps) {
+      const newProps = getNewChartProps(value, true, this.state.chartProps);
+      this.setState({ chartProps: newProps });
+    }
+  };
+
+  private onYSet = (value: ISelectorConfig): void => {
+    if (this.state.chartProps) {
+      const newProps = getNewChartProps(value, false, this.state.chartProps);
+      this.setState({ chartProps: newProps });
+    }
+  };
+
+  private selectPointFromChartLargeData = async (
+    data: IScatterPoint
+  ): Promise<void> => {
+    selectPointFromChartLargeData(
+      data,
+      this.setLocalCausalData,
+      this.toggleSelectionOfPoint,
+      this.props.onDataClick,
+      this.setTemporaryPointToCopyOfDatasetPoint,
+      this.props.telemetryHook
+    );
+  };
+
+  private setLocalCausalData = async (absoluteIndex: number): Promise<void> => {
+    this.setState({
+      isLocalCausalDataLoading: true,
+      localCausalErrorMessage: undefined
+    });
+    this.props.onDataClick(true, undefined);
+    const localCausalData = await getLocalCausalFromSDK(
+      this.props.causalId,
+      absoluteIndex,
+      this.context.requestLocalCausalEffects
+    );
+    if (localCausalData && !instanceOfLocalCausalData(localCausalData)) {
+      this.setState({
+        isLocalCausalDataLoading: false,
+        localCausalData: undefined,
+        localCausalErrorMessage: getErrorMessage(localCausalData)
+      });
+      return;
+    }
+    this.setState({
+      isLocalCausalDataLoading: false,
+      localCausalData: localCausalData as ICausalAnalysisData
+    });
+    this.props.onDataClick(false, localCausalData as ICausalAnalysisData);
+  };
+
+  private selectPointFromDropdown = (
+    _event: React.FormEvent<IComboBox>,
+    item?: IComboBoxOption
+  ): void => {
+    selectPointFromDropdownIntl(
+      this.setTemporaryPointToCopyOfDatasetPoint,
+      this.toggleSelectionOfPoint,
+      this.setLocalCausalData,
+      this.props.onDataClick,
+      item,
+      this.props.telemetryHook
+    );
+  };
+
+  private setTemporaryPointToCopyOfDatasetPoint = (
+    index: number,
+    absoluteIndex: number
+  ): void => {
+    this.setState({
+      temporaryPoint: getTemporaryPoint(
+        this.context.jointDataset,
+        index,
+        absoluteIndex
+      )
```

Comment:
looks like some of these files are going over the 300 line limit and will need to be refactored

Reply:
Only this file is going above 300 in this PR. I added a comment saying the same :) I am out of ideas and open to any ideas on reducing lines :)